### PR TITLE
[Issue-587] carbon table name should be case insensitive

### PR DESCRIPTION
**Pull request by foryou2030**
fix issue #587

### CODE REVIEW

**Comment by QiangCai** [comment]
@jackylk what's your opinion?

**Comment by gvramana** [comment]
LGTM

**Comment by Vimal-Das** [comment]
LGTM, @foryou2030  please rebase.

**Comment by foryou2030** [comment]
@Vimal-Das rebase done

**Comment by QiangCai** [comment]
please align with @eason-lyx because of duplicate PR #633

**Comment by eason-lyx** [comment]
oh,i will close,please @foryou2030 reference the PR to modify.

**Comment by eason-lyx** [comment]
link #632

File: integration/spark/src/test/scala/org/carbondata/spark/testsuite/dataretention/DataRetentionTestCase.scala [hunk 1]
```diff
@@ -154,4 +155,29 @@ class DataRetentionTestCase extends QueryTest with BeforeAndAfterAll {
...
+      CREATE TABLE IF NOT EXISTS carbon_TABLE_1
+      (ID Int, date Timestamp, country String,
+      name String, phonetype String, serialname String, salary Int)
+      STORED BY 'org.apache.carbondata.format'
+      TBLPROPERTIES('DICTIONARY_EXCLUDE'='country,phonetype,serialname',
+      'DICTIONARY_INCLUDE'='ID')
+      """)
+
+    sql("LOAD DATA LOCAL INPATH '" + resource +
+      "emptyDimensionData.csv' into table carbon_TABLE_1")
```
Comment (QiangCai) [inline integration/spark/src/test/scala/org/carbondata/spark/testsuite/dataretention/DataRetentionTestCase.scala]: please use diff case for table name in load, select, drop, delete sql

**Comment by QiangCai** [comment]
verified. LGTM.

**Pull request merged**